src/coprocessor/dag/rpn_expr/impl_op.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::types::RpnFnCallPayload;
+use crate::coprocessor::dag::expr::EvalContext;
+use crate::coprocessor::Result;
+
+#[derive(Debug, Clone, Copy)]
+pub struct RpnFnLogicalAnd;
+
+impl_template_fn! { 2 arg @ RpnFnLogicalAnd }
+
+impl RpnFnLogicalAnd {
+    #[inline]
+    fn call(
+        _ctx: &mut EvalContext,
+        _payload: RpnFnCallPayload<'_>,
+        arg0: &Option<i64>,
+        arg1: &Option<i64>,
+    ) -> Result<Option<i64>> {
+        // Intentionally not merging `None` and `Some(0)` conditions to be clear.
+        Ok(match arg0 {
```

Comment:
Maybe `match (arg0,arg1)` will be easier to review

Reply:
Good idea!